transform/models/marts/geo_reference/geo_reference__building_footprints_with_tiger.sql

```diff
@@ -0,0 +1,80 @@
+/* Columns selected and renamed from footprints, blocks, and places
+are reused in several places. To avoid repeating them with the possibility
+for errors, we set them in lists/dictionaries where appropriate. */
+{%- set footprint_cols = ['"release"', '"capture_dates_range"'] -%}
+
+{%- set block_cols = {
+      '"COUNTYFP20"': '"county_fips"',
+      '"TRACTCE20"': '"tract"',
+      '"BLOCKCE20"': '"block"',
+      '"GEOID20"': '"block_geoid"',
```

Comment:
@AeriShan-ODI I'm curious if you have patterns for this that you like: I didn't like duplicating the list of columns in several places as it's annoying to keep them in sync if you move or rename things. But maybe this is too-much-clever-Jjinja?

Reply:
what's making you feel like it may be too much too clever? and what's the alternative (if not going back to doing the join and dedupe in the marts models themselves)

Reply:
An alternative would be to just list out the column names every time I need them. It would be more repetition, but perhaps a bit easier to read. This is somewhat independent of whether the dedupe macro is a good idea

Reply:
Caveat: I don't know or really understand geospatial data. 

TBH, I do feel like it's "too-much-clever-Jjinja". I mean, it certainly is clever... but in these situations I always try to curb my own desire to be clever with how that gets looked at and maintained down the road. For example, it wasn't immediately obvious what you meant by deduplication - deduping records or columns (seems like both). The sort of abstraction that happened in the macro makes it hard to understand the operation and purpose. 
So with the caveat that I think you should keep this for the cleverness-as-interesting-and-educational factor, but with perhaps more descriptive documentation, my own approach in this would likely be to simply spell out the column names and use SELECT... EXCLUDE... or use a macro to only handle column name collisions. 

All this said, I'm a bit on the fence - we have the opportunity to push boundaries and be clever so why not? I'm so used to situations where cleverness ends up costing money

Reply:
> Caveat: I don't know or really understand geospatial data.

I'm a proponent of "spatial isn't special", i.e, we should use the same set of tools for working with spatial data, it's just another data type with some functions that know how to operate on it. All of which is to say, I think any code style and performance notes you have are fair game!

Here I'm writing this jinja-inflected SQL as if it were Python, and it probably doesn't help the legibility for people who are expecting SQL. I'll change it to just list the column names, there is probably enough jinja-weirdness going around that this is a bridge too far.

Reply:
I appreciate this discussion and the conclusion reached!